packages/context/src/binding.ts

```diff
@@ -239,11 +234,11 @@ export class Binding {
       if (this.scope === BindingScope.SINGLETON) {
         const ownerCtx = ctx.getOwnerContext(this.key);
         if (ownerCtx && this._cache.has(ownerCtx)) {
-          return this._cache.get(ownerCtx);
+          return this._cache.get(ownerCtx)!;
```

Comment:
I may have asked this before but what is the purpose of the trailing `!`?

Reply:
The `!` tells TS compiler that the value is not `undefined`. In this case, we already test the key exists in the cache 1st.